internal/utils/ci_providers.go

```diff
@@ -20,19 +20,20 @@ import (
 type providerType = func() map[string]string
 
 var providers = map[string]providerType{
-	"APPVEYOR":           extractAppveyor,
-	"TF_BUILD":           extractAzurePipelines,
-	"BITBUCKET_COMMIT":   extractBitbucket,
-	"BUDDY":              extractBuddy,
-	"BUILDKITE":          extractBuildkite,
-	"CIRCLECI":           extractCircleCI,
-	"GITHUB_SHA":         extractGithubActions,
-	"GITLAB_CI":          extractGitlab,
-	"JENKINS_URL":        extractJenkins,
-	"TEAMCITY_VERSION":   extractTeamcity,
-	"TRAVIS":             extractTravis,
-	"BITRISE_BUILD_SLUG": extractBitrise,
-	"CF_BUILD_ID":        extractCodefresh,
+	"APPVEYOR":            extractAppveyor,
+	"TF_BUILD":            extractAzurePipelines,
+	"BITBUCKET_COMMIT":    extractBitbucket,
+	"BUDDY":               extractBuddy,
+	"BUILDKITE":           extractBuildkite,
+	"CIRCLECI":            extractCircleCI,
+	"GITHUB_SHA":          extractGithubActions,
+	"GITLAB_CI":           extractGitlab,
+	"JENKINS_URL":         extractJenkins,
+	"TEAMCITY_VERSION":    extractTeamcity,
+	"TRAVIS":              extractTravis,
+	"BITRISE_BUILD_SLUG":  extractBitrise,
+	"CF_BUILD_ID":         extractCodefresh,
+	"CODEBUILD_INITIATOR": extractAwsCodePipelines,
```

Comment:
you need to compare the value of `CODEBUILD_INITIATOR` with StartWith `codepipeline`

Reply:
I think this should be enough, and it simplifies the code. Wdyt @andrea-mosk ?

Reply:
Yeah, unfortunately CodeBuild builds can be started by a variety of other AWS services. This won't be present only for CodePipeline, but also for all the other services, so we need to make sure it starts with `codepipeline`

Reply:
Should be fixed now :)